src/engine/strat_engine/metadata.rs

```diff
@@ -205,64 +212,46 @@ impl MDARegions {
 
         Ok(MDARegions {
             region_size: region_size,
-            mdas: [MDAHeader::new(), MDAHeader::new()],
-            latest_json: None,
+            mdas: [MDAHeader::new(per_region_size), MDAHeader::new(per_region_size)],
         })
     }
 
+    // Construct MDARegions based on on-disk info
     pub fn load(header: &StaticHeader, f: &mut File) -> EngineResult<MDARegions> {
         let region_size = Sectors(*header.mda_size / NUM_MDA_REGIONS);
+        let per_region_size: u64 = *region_size * SECTOR_SIZE;
 
-        let mut load_region = |region: u8| -> EngineResult<(MDAHeader, Option<Vec<u8>>)> {
+        let mut load_a_region = |region: u8| -> EngineResult<MDAHeader> {
             let mut hdr_buf = [0u8; MDA_REGION_HDR_SIZE];
             let offset = BDA_STATIC_HDR_SIZE + (region as u64 * *region_size * SECTOR_SIZE);
 
             try!(f.seek(SeekFrom::Start(offset)));
             try!(f.read_exact(&mut hdr_buf));
+            let mda = try!(MDAHeader::from_buf(&hdr_buf, per_region_size));
 
-            let (mda, json_crc) = try!(MDAHeader::from_buf(&hdr_buf));
-            if mda.used > *region_size * SECTOR_SIZE {
-                return Err(EngineError::Engine(ErrorEnum::Invalid,
-                                               "mda0.used > region_size".into()));
-            }
+            // Loading checks json CRC
+            try!(mda.load_region(f));
 
-            let json_buf = {
-                if mda.used == 0 {
-                    None
-                } else {
-                    let mut json_buf = vec![0u8; mda.used as usize];
-                    try!(f.read_exact(&mut json_buf));
-
-                    if json_crc != crc32::checksum_ieee(&json_buf) {
-                        return Err(EngineError::Engine(ErrorEnum::Invalid,
-                                                       "MDA region json CRC".into()));
-                    }
-                    Some(json_buf)
-                }
-            };
-
-            Ok((mda, json_buf))
+            Ok(mda)
         };
 
-        let (mda0, json0) = load_region(0)
-            .unwrap_or_else(|_| load_region(2).unwrap_or_else(|_| (MDAHeader::new(), None)));
-        let (mda1, json1) = load_region(1)
-            .unwrap_or_else(|_| load_region(3).unwrap_or_else(|_| (MDAHeader::new(), None)));
-
-        let latest_json = match mda0.last_updated.cmp(&mda1.last_updated) {
-            Ordering::Less => json0,
-            Ordering::Greater => json1,
-            Ordering::Equal => json0,
-        };
+        let mda0 =
```

Comment:
Still wondering if you should do some fixing up if you fail on one of the two copies.